Dockerfile

```
@@ -0,0 +1,18 @@
+FROM ruby:2.6.3
+RUN apt-get update -qq && apt-get install -y nodejs postgresql-client
+RUN mkdir /myapp
+WORKDIR /myapp
+COPY Gemfile /myapp/Gemfile
```

Why do you copy the Gemfile, install and run bundle and then copy the rest of the project. Wouldn't be easier just to copy everything and then install and run bundle?

---

I thought the same thing, it would be easier to script. 

I'm trying to find out why every article/doc that I find has this convention. My guess is that it's a _fail fast_ practice, where you first check if every dependency can be installed and, if everything gets to install correctly, then you copy the entire directory (which may take a considerable amount of time if the project is big enough).

[Here's another guess that I found on StackOverflow.](https://stackoverflow.com/questions/50083406/why-would-you-copy-your-gemfile-lock-into-docker-then-immediately-overwrite-it)

---

It doesn't make a lot of sense to me but if it's in the official documentation... Just be aware that each RUN, COPY and ADD command creates a new layer:

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/